my_list.rb

```diff
@@ -0,0 +1,11 @@
+class MyList
+  include Enumerable
+
+  def initialize(*list)
+    @list = list
+  end
+
+  def each
+    @list.each { |item| yield(item) if block_given? }
+  end
```

Comment:
- [ ] As requested earlier, Kindly consider creating the `each` method on your own instead of using the default `each` method.

###  Here is an example

```
def each
    index = 0
    while index < @list.length
      yield(@list[index]) if block_given?
      index += 1
    end
  end

```